### PR TITLE
Importing locations

### PR DESCRIPTION
**Pull request by milank7**

### CODE REVIEW

File: src/main/java/com/revealprecision/revealserver/service/PlanLocationsService.java [hunk 1]
```diff
@@ -60,26 +66,53 @@ public PlanLocations getPlanLocationByPlanIdentifierAndLocationIdentifier(UUID p
...
+
+      Set<UUID> currentLocation = plan.getPlanLocations().stream()
+          .map(planLocations1 -> planLocations1.getLocation().getIdentifier()).collect(
+              Collectors.toSet());
+
+      Set<UUID> locationsToAdd = new HashSet<>(locations);
+      locationsToAdd.removeAll(currentLocation);
+      currentLocation.removeAll(locations);
+
+      List<Location> addLocaitons = locationService.getAllByIdentifiers(new ArrayList<>(locationsToAdd));
```
Comment (trevlenb2) [inline src/main/java/com/revealprecision/revealserver/service/PlanLocationsService.java:84]: misspell -> addLocaitons

File: src/main/java/com/revealprecision/revealserver/service/PlanLocationsService.java [hunk 2]
```diff
@@ -60,26 +66,53 @@ public PlanLocations getPlanLocationByPlanIdentifierAndLocationIdentifier(UUID p
...
-    });
+      if(currentLocation.size() > 0) {
+        planLocationsRepository.deletePlanLocationsByPlanAndLocation(planIdentifier, new ArrayList<>(currentLocation));
+      }
+
+
+//      planLocations.forEach(el -> {
+//        if (!locations.contains(el.getLocation().getIdentifier())) {
+//          planLocationsRepository.delete(el);
+//        } else {
```
Comment (trevlenb2) [inline src/main/java/com/revealprecision/revealserver/service/PlanLocationsService.java:100]: remove comment?

File: src/main/java/com/revealprecision/revealserver/service/PlanService.java [hunk 3]
```diff
@@ -38,6 +41,9 @@
   private final LookupInterventionTypeService lookupInterventionTypeService;
   private final TaskService taskService;
 
+  @Autowired
+  private KafkaTemplate<String, Message> kafkaTemplate;
```
Comment (trevlenb2) [inline src/main/java/com/revealprecision/revealserver/service/PlanService.java:45]: Constructor injection as per current standard?

File: src/main/resources/db/migration/V0_0_1__initial_migration.sql [hunk 4]
```diff
@@ -589,7 +589,7 @@ CREATE TABLE IF NOT EXISTS location_relationship
     location_hierarchy_identifier UUID                     NOT NULL,
     location_identifier           UUID                     NOT NULL,
     parent_identifier             UUID,
-    ancestry                      UUID[]                   NOT NULL,
+    ancestry                      UUID[]                   ,
```
Comment (trevlenb2) [inline src/main/resources/db/migration/V0_0_1__initial_migration.sql:592]: how come ancestry is optional?

Comment (milank7) [reply src/main/resources/db/migration/V0_0_1__initial_migration.sql:592]: We didn't use this column anywhere, so I was talking with Linda to maybe remove it from db, so first I delete not_null constraint to see is there going to be an exception if it is null.. so maybe we can remove it now

**Review by trevlenb2 — APPROVED** [review]

**Pull request merged**